src/main/java/net/sf/mpxj/ManuallyScheduledTaskCalendar.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * file:       ProjectCalendar.java
+ * author:     Fabian Schmidt
+ * date:       16/08/2024
+ */
+
+/*
+ * This library is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation; either version 2.1 of the License, or (at your
+ * option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+ * License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+ */
+
+package net.sf.mpxj;
+
+import java.time.DayOfWeek;
+import java.time.LocalDate;
+import java.time.LocalTime;
+
+/**
+ * This class represents a Calendar Definition record for an MPP Manually Scheduled task.
+ */
+public class ManuallyScheduledTaskCalendar extends ProjectCalendar
+{
+   public ManuallyScheduledTaskCalendar(ProjectCalendar calendar, ResourceAssignment assignment)
+   {
+      super(calendar.getParentFile(), true);
+      m_calendar = calendar;
+      m_assignment = assignment;
+      m_assignment_start_date = m_assignment.getStart().toLocalDate();
+      m_assignment_end_date = m_assignment.getFinish().toLocalDate();
+   }
+
+   /**
+    * Retrieves the working hours on the given date.
+    *
+    * @param date required date
+    * @return working hours
+    */
+   @Override protected ProjectCalendarHours getRanges(LocalDate date)
+   {
+      ProjectCalendarHours effectiveRanges = m_calendar.getRanges(date);
```

Comment:
Looking very good so far. I've run this branch against my test data and I'm seeing changes in the output that make sense. The one issue I've encountered is that I have an MPP file with an apparently completely non-working calendar, so `effectiveRanges` here is empty, and after this `if (effectiveRanges.isEmpty() && (date.equals(m_assignment_start_date) || date.equals(m_assignment_end_date)))` statement, `effectiveRanges` is still empty.

It looks like we need something like:

```java
if (effectiveRange.isEmpty())
{
   return effectiveRanges;
}
```

at line 93 to exist gracefully in this situation - what do you think?

Reply:
My Project refused to create such calendar with the following message:
![You must have at least one working day in the calendar.](https://github.com/user-attachments/assets/2c0ed7a6-0369-4e35-9b44-062d208171cd)
But I am fine with adding this handling in case of.

Reply:
Done. Took two tries.